### ISSUE TITLE
Placeholder here for Boundary Page Wireframe

### ISSUE THREAD

**Issue by meghaklp** [klpdotorg/dubdubdub#473]
![boundarysschoolpage](https://cloud.githubusercontent.com/assets/465382/7133940/d5337daa-e2b5-11e4-826a-b8956b14ce6b.png)

![boundaryspreschoolpage](https://cloud.githubusercontent.com/assets/465382/7133934/c77e942e-e2b5-11e4-982f-4afa13492b20.png)

**Comment by gkjohn** [comment]
@meghaklp This looks great. A few thoughts:
1. To be able to select the year? 
2. The "Compare To" button is there but just to make sure that works. And a page for that comparison too? Only for like entities? Or any two entities? Eg: Block to Blocks only or Blocks to Clusters/Districts/Schools etc. too? [Consider this a question.]

**Comment by meghaklp** [comment]
1.  Could add an year selector, except then perhaps the school page should also have an year selector? @batpad , thoughts?
2.  I need to give this a bit more thought. The comparison, especially with programme data makes far more sense. We could just compare number of schools, students, etc, up front anyway.

**Comment by batpad** [comment]
We eventually would want year selectors, yes. This is *probably* not going to happen in the first iteration of the boundary pages, but definitely something soon as the database now supports it. The problem is, that we need to restructure some things since I'd want to get the DISE yearly data directly from the DISE app API since it doesn't make sense to duplicate that functionality in the dubdubdub app .. 

Anyways, I think a year selector is probably good to factor into the design, just with knowing that we will probably hide it for a first iteration. Will that be okay?

Again, compare is probably not going to work in the first iteration for boundary level things. Can we discuss this tomorrow? Do we want to be able to compare across entity types, etc.. ? This would factor into both the design and implementation. This would be lots of fun to do once we have the APIs up, probably not in first iteration tho - lets definitely discuss this tomorrow though!

**Comment by gkjohn** [comment]
1. Wherever data is available year-wise and wherever data is available from multiple sources, it makes sense to have a selector. 
2. See the mockups Uncommon did? I think it has it. @meghaklp

**Comment by batpad** [comment]
I think these are the Uncommon mockups for boundary level pages: http://klp-demo.aws.af.cm/cluster.php  .. don't think there's a year selector there? @gkjohn were you thinking of the Compare modal mockups, which do have year selectors?

In general, we are going to think of how to integrate year selectors across school pages, boundary pages as well as the compare modal once we have that functionality enabled in the API. Let's discuss this tomorrow so that we're all on the same page with the plan for this.

**Comment by gkjohn** [comment]
Ah yes. Though my point still holds. If there is data for multiple years, show it.

**Comment by batpad** [comment]
Absolutely :) - that is most definitely the idea. Steps for implementation is like this though:

 - first, get new data into the system
 - aggregate this data at boundary levels and expose API end-points for the same
 - allow querying by academic year in the API

Since we're still at step 1 of this process, I just want to make sure we can get something out the door in terms of front-end before we have academic year filtering working in the API, and then go back and add that to the front-end once we have the ability at the API level.

**Comment by meghaklp** [comment]
1) So, every dashboard/ report/ info page of any sort on the website can
have an academic year selector, correct?
2) And then there's the additional complexity we can throw in that
comparison can happen across years also?
@gkjohn - Do we want the latter?

We probably need some uniform selector across these pages, so the user gets
familiar with using the switch. Just a place holder for now.

Thanks,
Megha
9845844936 | www.klp.org.in

**Comment by gkjohn** [comment]
1) Yes. 
2) Yes. But not a priority.

**Comment by meghaklp** [comment]
I'm just wondering  - what's the use case really for someone looking at the status/data of a school / boundary for a previous year, unless they can look at it in relation to the current year. 

I may for example want to see if the number of schools / students dropped in comparison. So, does the year selector work in conjunction with a comparison...

**Comment by meghaklp** [comment]
![boundarysschoolpage](https://cloud.githubusercontent.com/assets/465382/7250921/e3108e48-e843-11e4-8f68-3186b363c7ed.png)
![boundaryspreschoolpage](https://cloud.githubusercontent.com/assets/465382/7250922/e3108b64-e843-11e4-8315-e623a2a33a98.png)

**Comment by gkjohn** [comment]
@meghaklp Going to be a pedant. The School/Preschool tab/choice won't work if the hierarchy is D/B/C to D/P/C. Will work if PIN Code or MP/MLA etc. or Revenue District or GP/ZP

**Comment by meghaklp** [comment]
Pendant? Means?... We could show tabs only when there is data for both school & preschool. The other option is putting preschool data below school data. These designs are with print in mind btw. And my philosophy was show KLP data primarily and DISE data additionally. So, KLP data comes first on the right. Where we don't have KLP data, we can't help but just show DISE

**Comment by gkjohn** [comment]
@meghaklp "a person who annoys other people by correcting small errors and giving too much attention to minor details" :see_no_evil: 
When will you put pre-school data below the school data or vice-verse? When at an aggregation other than D/B/C etc?

**Comment by meghaklp** [comment]
Urban slang is it? I even looked it up before I replied ... And yes, one below the other or tabs only where relevant.

**Comment by gkjohn** [comment]
Nope! "From Middle French pedant, pedante, from Italian pedante (“a teacher, schoolmaster, pedant”), of uncertain origin, traced by some sources to Latin paedagogans, present participle of paedagogare ( = to teach, from Greek "paedagogein" = to instruct children ). Confer French pédant."

**Comment by meghaklp** [comment]
Ooopsie.... I read it as "peNdant"... :open_mouth:

**Comment by gkjohn** [comment]
:sob:

**Comment by meghaklp** [comment]
@batpad  - the map on top will need the boundary marker and I guess we have those...Is that okay to show @gkjohn? Just the marker, ie., we can't do much else I suspect.

**Comment by gkjohn** [comment]
@meghaklp We do have those. If not, we can petition @geohacker to create centroids to show them.

**Comment by meghaklp** [comment]
I saw cluster markers, I thought and also that @geohacker has been there, done that.

**Comment by batpad** [comment]
Yup, we have markers ..

**Comment by meghaklp** [comment]
![boundary-school](https://cloud.githubusercontent.com/assets/465382/7413705/82a046a0-ef6b-11e4-9485-2900f6e54275.png)
![boundary-preschool](https://cloud.githubusercontent.com/assets/465382/7413706/82c98c4a-ef6b-11e4-806b-497f1967520a.png)

**Comment by meghaklp** [comment]
@batpad  - the map is an eye-wash right now, uses some one particular school's lat long and feeds in the admin data.  Which is why even for Preschool the admin data is the same.
I found the tab.js to be very complex, it's very closely tied to the school page, and i couldn't figure what to keep and what to throw out. Tab is only necessary for the MP, MLA, Wards, We can place preschool data below school data for that. But if we decide to go with tabs, I'll need you to help with that Sanjay.

**Comment by meghaklp** [comment]
Compare piece remains. Will work on it later.

**Comment by batpad** [comment]
@meghaklp yes, I've been meaning to refactor tabs.js. With a little bit of refactoring, it could be used elsewhere, but it also fulfils some very particular use-cases on the school page which I don't think we need here (handling url, caching of data, making requests when needed, stuff like this).

Here I'd just query the API and render the templates, should be relatively straightforward.

Yea, the compare.js is a bit of a mess, I need to look at it a bit carefully and see how best to integrate comparing boundaries in there .. will be nice to know the overview, though, of what all needs to be compared - i.e. do we allow comparing a boundary to a school - a boundary / school to itself across years? etc.. Let know if you want to discuss that / make a separate ticket for it.

Any-how, this looks great! Will check-out the branch, etc. Anyways, we're still probably a bit away from having all the API bits in place (switching years, and boundary aggregations), so no rush on the compare. Would probably be nice to conceptualize compare so that we can use the same design + code on boundary page, school page as well as on the map ..

Thanks!

**Comment by meghaklp** [comment]
@iambibhas  - The branch is called boundary-page

**Comment by batpad** [comment]
So some notes questions from our call earlier:

1. To be aware that we don't have yearly data for no. of schools. Seems like this we keep as is, and change if required / based on feedback

2. General question / feature request to have more navigation from this page / perhaps a way to look at a list of schools within this district.

3. The Average Enrolment by Category chart not being a percentage value maybe confusing.

4. Preschool infrastructure data may not be available for all schools within boundary. Front-end should probably make that clear some-how.

**Comment by meghaklp** [comment]
Pop up is on the branch boundary-pop-up, @batpad @iambibhas 

![image](https://cloud.githubusercontent.com/assets/465382/8828040/ad2e8188-30ad-11e5-9cee-999cf1cd1372.png)

**Comment by meghaklp** [comment]
The only thing that maybe nice is making that 'C' an 'SC'. I have to define a new css class for that for the map, because the circle can take only 1 letter without becoming an ellipse. So, i'll leave it for later.

**Comment by gkjohn** [comment]
No number of schools on the popup? 

Sent from Outlook

    _____________________________

**Comment by meghaklp** [comment]
Damn. I forgot all about it in the joy of getting this pop up to not get half hidden. So, now I'll need some help from you guys to position the marker on the map layers such that this box doesn't go hidden. Think it's some js voodoo. @batpad - Look at it closer to the end of sprint maybe.

![image](https://cloud.githubusercontent.com/assets/465382/8830261/ec73837e-30b8-11e5-9aa6-1d1fae06004b.png)
